inc/query/query.php

```diff
@@ -24,15 +29,17 @@ public function __construct( $args ) {
 	 *
 	 * @return mixed
 	 */
-	public function alter_clauses( &$clauses, $id_column, $pass_thru_order = false ) {
+	public function alter_clauses( &$clauses, $table, $column, $pass_thru_order = false ) {
+		$this->table = $table;
+		$id_column = "$table.$column";
 		$this->handle_query_join( $clauses, $id_column, $pass_thru_order );
 
 		if ( empty( $this->args['relation'] ) && ! empty( $this->args['sibling'] ) ) {
 			$this->handle_query_sibling( $clauses, $id_column );
 		}
 
 		global $wpdb;
-		$clauses['groupby'] = empty( $clauses['groupby'] ) ? "$wpdb->posts.ID" : "{$clauses['groupby']}, $wpdb->posts.ID";
+		$clauses['groupby'] = empty( $clauses['groupby'] ) ? "$id_column" : "{$clauses['groupby']}, $id_column";
```

Comment:
I think passing the whole `id_column` is still valid. We just need to make it group by `id_column` instead of `$wpdb->posts.ID`.

Reply:
man I totally overlooked that as a simple solution!

Reply:
Awesome!